applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_prebuckling_solver.py

```diff
@@ -0,0 +1,114 @@
+from __future__ import print_function, absolute_import, division  # makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+
+# Importing the Kratos Library
+import KratosMultiphysics
+
+# Import applications
+import KratosMultiphysics.StructuralMechanicsApplication as StructuralMechanicsApplication
+
+# Import base class file
+from KratosMultiphysics.StructuralMechanicsApplication.structural_mechanics_solver import MechanicalSolver
+
+from KratosMultiphysics import eigen_solver_factory
+from KratosMultiphysics import python_linear_solver_factory as linear_solver_factory
+import KratosMultiphysics.kratos_utilities as kratos_utils
+
+def CreateSolver(main_model_part, custom_settings):
+    return PrebucklingSolver(main_model_part, custom_settings)
+
+class PrebucklingSolver(MechanicalSolver):
+    """The structural mechanics prebuckling solver.
+
+    This class creates the mechanical solvers for prebuckling analysis.
+
+    See structural_mechanics_solver.py for more information.
+    """
+    def __init__(self, main_model_part, custom_settings):
+        # Construct the base solver.
+        super(PrebucklingSolver, self).__init__(main_model_part, custom_settings)
+        self.solverFlag = "eigen"
+        KratosMultiphysics.Logger.PrintInfo("::[PrebucklingSolver]:: ", "Construction finished")
+
+    @classmethod
+    def GetDefaultSettings(cls):
+        this_defaults = KratosMultiphysics.Parameters("""{
+            "buckling_settings"     : {
+                "initial_step"          : 1.0,
```

Comment:
initial/small/big do not describe very well what these numbers are used for. Maybe you can find a better naming.